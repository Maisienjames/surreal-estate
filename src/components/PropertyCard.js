import React from "react";
import PropTypes from "prop-types";
import "../styles/property-card.css";

const PropertyCard = (props) => {
  const { title, type, bathrooms, bedrooms, price, city, email } = props;

  return (
    <div className="property-card">
      <div className="property-card__title">{title}</div>
      <div className="property-card__type">{type}</div>
      <div className="property-card__bathrooms">{bathrooms}</div>
      <div className="property-card__bedrooms">{bedrooms}</div>
      <div className="property-card__price">{price}</div>
      <div className="property-card__city">{city}</div>
      <div className="property-card__email">{email}</div>
    </div>
  );
};

export default PropertyCard;

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  bathrooms: PropTypes.number.isRequired,
  bedrooms: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  city: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
};
