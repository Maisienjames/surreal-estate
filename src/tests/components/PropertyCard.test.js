/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "Stub title",
    type: "Stub type",
    bathrooms: 11,
    bedrooms: 5,
    price: 1111111,
    city: "Manchester",
    email: "joebloggs@gmail.com",
  };

  test("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders correct values for props", () => {
    const { getByText } = render(
      <PropertyCard
        title={validProps.title}
        type={validProps.type}
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        price={validProps.price}
        city={validProps.city}
        email={validProps.email}
      />
    );
    expect(getByText("Stub title")).toHaveClass("property-card__title");
    expect(getByText("Stub type")).toHaveClass("property-card__type");
    expect(getByText(11)).toHaveClass("property-card__bathrooms");
    expect(getByText(5)).toHaveClass("property-card__bedrooms");
    expect(getByText(1111111)).toHaveClass("property-card__price");
    expect(getByText("Manchester")).toHaveClass("property-card__city");
    expect(getByText("joebloggs@gmail.com")).toHaveClass("property-card__email");
  });
});
