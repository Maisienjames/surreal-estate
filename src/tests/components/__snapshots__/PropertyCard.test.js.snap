// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="property-card__title"
    >
      Stub title
    </div>
    <div
      class="property-card__type"
    >
      Stub type
    </div>
    <div
      class="property-card__bathrooms"
    >
      11
    </div>
    <div
      class="property-card__bedrooms"
    >
      11
    </div>
    <div
      class="property-card__price"
    >
      1111111
    </div>
    <div
      class="property-card__city"
    />
    <div
      class="property-card__email"
    />
  </div>
</DocumentFragment>
`;
